src/main/kotlin/no/nav/familie/ef/sak/beregning/BeregningUtils.kt

```diff
@@ -2,72 +2,98 @@ package no.nav.familie.ef.sak.beregning
+                                                                  inntekt = inntekt,
+                                                                  grunnbeløp = it.beløp))
         }
+    }
+
+    private fun beregnAvkortning(grunnbeløp: BigDecimal, inntekt: BigDecimal): BigDecimal {
+        val inntektOverHalveGrunnbeløp = inntekt.subtract(grunnbeløp.multiply(BigDecimal(0.5)))
+        return if (inntektOverHalveGrunnbeløp > BigDecimal.ZERO)
+            inntektOverHalveGrunnbeløp.multiply(REDUKSJONSFAKTOR).setScale(5, RoundingMode.HALF_DOWN) else BigDecimal.ZERO
+    }
+
 
+    fun indeksjusterInntekt(sisteGrunnbeløpsdato: LocalDate,
+                            inntekter: List<Inntektsperiode> = emptyList()): List<Inntektsperiode> {
 
-        private fun beregnAvkortning(grunnbeløp: BigDecimal, inntekt: BigDecimal): BigDecimal {
-            val inntektOverHalveGrunnbeløp = inntekt.subtract(grunnbeløp.multiply(BigDecimal(0.5)))
-            return if (inntektOverHalveGrunnbeløp > BigDecimal.ZERO)
-                inntektOverHalveGrunnbeløp.multiply(REDUKSJONSFAKTOR).setScale(5, RoundingMode.HALF_DOWN) else BigDecimal.ZERO
+        val sistBrukteGrunnbeløp = finnGrunnbeløp(sisteGrunnbeløpsdato)
+        if (sisteGrunnbeløp == sistBrukteGrunnbeløp) {
+            return inntekter
         }
 
-        fun finnStartDatoOgSluttDatoForBeløpsperiode(beløpForInnteksperioder: List<Beløpsperiode>,
-                                                     vedtaksperiode: Periode): List<Beløpsperiode> {
-            return beløpForInnteksperioder.mapNotNull {
-                when {
-                    it.periode.omsluttesAv(vedtaksperiode) -> {
-                        it
-                    }
-                    it.periode.overlapperIStartenAv(vedtaksperiode) -> {
-                        it.copy(periode = it.periode.copy(fradato = vedtaksperiode.fradato))
-                    }
-                    vedtaksperiode.overlapperIStartenAv(it.periode) -> {
-                        it.copy(periode = it.periode.copy(tildato = vedtaksperiode.tildato))
-                    }
-                    vedtaksperiode.omsluttesAv(it.periode) -> {
-                        it.copy(periode = it.periode.copy(fradato = vedtaksperiode.fradato, tildato = vedtaksperiode.tildato))
-                    }
-                    else -> {
-                        null
-                    }
-                }
+        return inntekter.flatMap { justerInntektsperiode(it, sistBrukteGrunnbeløp) }
+
+    }
+
+    private fun justerInntektsperiode(inntektsperiode: Inntektsperiode,
+                                      sistBrukteGrunnbeløp: Grunnbeløp): List<Inntektsperiode> {
+        val (startDato, sluttDato, inntekt, samordningefradrag) = inntektsperiode
+        return finnGrunnbeløpsPerioder(startDato, sluttDato).map {
+            if (it.periode.fradato > sistBrukteGrunnbeløp.fraOgMedDato && it.beløp != sistBrukteGrunnbeløp.grunnbeløp) {
```

Comment:
Når vi g-omregner så trenger vi vel å avkorte perioder og innteker fra gitt dato, sånn att når vi sender inn datoer fra mai, sånn att alle innteker fra mai burde vel lønnsjusteres? 
```suggestion
            if (it.periode.fradato >= sistBrukteGrunnbeløp.fraOgMedDato && it.beløp != sistBrukteGrunnbeløp.grunnbeløp) {
```

Reply:
Enig i at vi må ha >= her - for den vil vel kuttes i [mars-april] og [mai, juli] hvis man har periode mars-juli som skal g-omregnes i mai

Reply:
it.periode.fradato er datoen for forrige brukte grunnbeløp. Den perioden er justert allerede. Jeg har lagt til en tyest til og endret navn på it, for å gjøre det tydligere.